### ISSUE TITLE
Certain Audio has static sounds

### ISSUE THREAD

**Issue by Supertacoz** [bluemagic123/tModLoader#3104]
### Version

Latest 1.4-stable

### OS

Linux

### Platform

Steam

### This bug affects

Gameplay as a Player

### Description

Whenever I load up tmodloader, it starts completely normally, but if I listen closely in the menu, I hear a bit of sharp static noise. this noise becomes slightly more noticeable when in a world/save, and quickly causes me to get a headache.

### Log File

[client.log](https://github.com/tModLoader/tModLoader/files/10072053/client.log)

### Steps to reproduce

1. Be on linux
2. launch tmodloader
3. listen, maybe go into a save and listen more.

### Expected Behavior

audio should be playing normally with no sharp static noise.

### Actual Behavior

sharp static noise occurs.

### Reproduction frequency

every time I use tmodloader.

### Additional Information

Unsure if this is important, but I am using linux mint with Linux kernel 6.0, and mesa driver version 22.2.4.

**Comment by JavidPack** [comment]
Aside from some bugs with WingSlotExtra, I'm not seeing anything that would indicate a reason. We've recently updated all music to work the same as vanilla terraria on all platforms. Can you confirm if you notice similar things on Terraria?

**Comment by Supertacoz** [comment]
Tested the latest version of the base game (1.4.4.9) and found the issue is not present there.

**Comment by Supertacoz** [comment]
upon retesting Tmodloader, I found that the static noise has decreased in volume, will continue testing.

**Comment by Supertacoz** [comment]
I tried disabling three different mods on their own,
-wing slot extra is not the culprit
-calamity music mod is not the culprit
-infernum mode is not the culprit
I am unsure what could be wrong.

**Comment by JavidPack** [comment]
You should make sure that it still happens with 0 mods enabled too, not just individually disabling mods.

**Comment by Supertacoz** [comment]
I can confirm it still occurs with no mods. would you like the modless log?

**Comment by JavidPack** [comment]
yes. Do note that we currently don't know why this is happening for you, Hopefully if others have this issue we can narrow down things by comparing computer settings

**Comment by Supertacoz** [comment]
[client.log](https://github.com/tModLoader/tModLoader/files/10079231/client.log)

**Comment by Supertacoz** [comment]
strangely, the audio issue has now disappeared. I have not changed anything, so I am unsure what happened.

**Comment by Supertacoz** [comment]
After having launched and played tmodloader a number of times more, the issue has reduced volume, and significantly reduced occurances, making it almost unnoticable now.

**Comment by Destructor-Ben** [comment]
I have an issue similar to this, however it is only on the stable version of tML, and if I use headphones it goes away.

**Comment by Supertacoz** [comment]
I am on the stable version of TML. I will check if headphones solve the issue.

**Comment by Supertacoz** [comment]
I tested it with headphones, and the issue is not present with them on. in addition, I found that it is only the music causing this, as if I turn music volume to zero in game, it stops, while still occurring if I set sound or ambience to zero.

edit: it is not always caused by the music apparently

**Comment by Supertacoz** [comment]
I also just tried tmodloader with proton 7.0-4, and the game worked, albeit there were several issues. 
1. the config menu of some mods were in chinese or japanese
2. once in game, everything except the hud was pure black unless I set the color scheme to retro or trippy. 

and now after using the proton version and switching back to the normal version, tmodloader now refuses to start, even after restarting steam.

Edit: NVM, fixed by deleting everything in the tmodloader folder. static is still present though.

**Comment by Supertacoz** [comment]
Here is the launch and native logs, as I noticed they had things that might show what the problem is.

[Launch.log](https://github.com/tModLoader/tModLoader/files/10116545/Launch.log)

[Natives.log](https://github.com/tModLoader/tModLoader/files/10116546/Natives.log)

**Comment by jeremyram777** [comment]
I have the exact same issue, and also noticed turning off game sound seems to fix it...which sucks. The only mods i have are calamity and the calamity music, disabling them does nothing.
did you find a solution?
Im on PC/windows and latest version of tmodloader

**Comment by Supertacoz** [comment]
I'm on a linux PC, and alas the issue still persists.

**Comment by ArgentumRhodon** [comment]
I am also having this issue.

**Comment by Supertacoz** [comment]
Unfortunately this issue continues into Tmodloader 1.4.4.

**Comment by Supertacoz** [comment]
I have managed to fix the issue! it is a linux specific issue, but it has to do with my audio control software, called pulse. 
If anyone else on linux has a similar crackling issue, Here is how to fix it (if you use pulse audio): 
go to /etc/pulse/default.pa 
look for a line that says load-module module-udev-detect
change it to load-module module-udev-detect tsched=0
Then save and exit the text document.
Next, go to /etc/modprobe.d/alsa-base.conf
at the very bottom of the file, add a new line: options snd-hda-intel vid=8086 pid=8ca0 snoop=0
Save the text document, then restart your PC. This should fix the audio crackling.

**Comment by qFelixDev** [comment]
This also works on Arch Linux. Thanks a lot!
